docs/source/release_notes/release-notes.rst

```diff
@@ -42,6 +42,22 @@ develop
 =======
 
 .. replace this with the release date and the above with v<tag> (the v makes the permalinks nice)
+
+.. list-table::
+    :widths: 5 40
+    :header-rows: 1
+
+    *    - Type
+         - Change
+
+    *    - |fixed|
+         - Snapshot transaction is now guaranteed to return getRowsColumnRange results in the correct order.
+           Previously while paging over row dynamic columns, if uncommitted or aborted transaction data was
```

Comment:
why would aborted data ever show up at all?

Reply:
@j-baker

Reply:
When we read data from the KVS, we scan backwards from the maximum possible Atlas timestamp until we find a value - this is how reads and conflict detection work. Transactions necessarily write their data before they commit, so KVS reads can return data that belongs to another transaction, which could be committed or uncommitted. Aborting a transaction means marking in the _transactions table that it did not (and cannot) commit, but it's not possible to delete all the data written by that transaction at the time of the abort for a variety of reasons (don't know all the places that transaction wrote, all necessary KVS nodes may not be up), so transactions in general may happen across invalid data. 

As implemented, whenever an invalid value is found, we do another batch of reads to continue scanning for a valid value at a lower timestamp. Unfortunately, any value where we had to do additional iterative lookups ended up behind the other results.